extensions/gitpod-web/src/extension.ts

```diff
@@ -399,86 +506,181 @@ export class GitpodWorkspaceTreeDataProvider implements vscode.TreeDataProvider<
 			const toClean = new Set<number>(this.ports.ports.keys());
 			const portsList = this.portStatus.getPortsList();
 			for (const portStatus of portsList) {
-				const currentStatus = portStatus.toObject();
-				const { name, localPort, description, exposed, served } = currentStatus;
+				const localPort = portStatus.getLocalPort();
+				const tunnel = this.tunnelMap.get(localPort);
```

Comment:
it is something new, why do we do it?

Reply:
is it because we have 2 duplicated structures, so we have 2 GitpodWorkspacePort for 2 ports? Why don't introduce PortsModel decoupled from presentations?

Reply:
`GitpodWorkspacePort` is a `PortModel` now, only provide data that `treeView` and `webview` need